### PR TITLE
Fix explanation of Cargo's behavior

### PR DESCRIPTION
**Pull request by steveklabnik**
https://github.com/rust-lang/rust/pull/25080/files#r29634986

r? @huonw

### CODE REVIEW

File: src/doc/trpl/guessing-game.md [hunk 1]
```diff
@@ -422,12 +422,17 @@ figures out all of the versions that fit your criteria, and then writes them
...
+until we explicitly upgrade, and so will anyone who we share our code with,
+thanks to the lock file.
 
-What about when we _do_ want to use `v0.4.0`? Cargo has another command,
+What about when we _do_ want to use `v0.3.9`? Cargo has another command,
 `update`, which says ‘ignore the lock, figure out all the latest versions that
 fit what we’ve specified. If that works, write those versions out to the lock
-file’.
+file’. But, by default, Cargo will only look for versions larger than `0.3.0`
+and smaller than `0.4.0`. If we want to move to `0.4.x`, we’d have to update
```
Comment (huonw) [inline src/doc/trpl/guessing-game.md]: Worth mentioning semver/breaking changes in here somewhere?

**Comment by huonw** [comment]
r=me once the comment has been considered (doesn't have to be accepted).

File: src/doc/trpl/guessing-game.md (hunk 1 above)
Comment (steveklabnik) [inline src/doc/trpl/guessing-game.md]: I'm not 100% sure. I'm so used to explaining every detail that this higher-level, ignore-the-details stuff is hard for me. I'm still trying to figure out what belongs and what doesn't.

Let's consider adding this kind of thing once the API evolution RFC lands.

**Pull request merged**